src/System.Drawing.Common/tests/mono/System.Drawing/BitmapTests.cs

```diff
@@ -0,0 +1,1762 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+//
+// Bitmap class testing unit
+//
+// Authors:
+// 	Jordi Mas i Hernàndez (jmas@softcatala.org>
+//	Jonathan Gilbert <logic@deltaq.org>
+//	Sebastien Pouliot  <sebastien@ximian.com>
+//
+// (C) 2004 Ximian, Inc.  http://www.ximian.com
+// Copyright (C) 2004,2006-2007 Novell, Inc (http://www.novell.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Security.Cryptography;
+using System.Security.Permissions;
+using System.Text;
+using System.Xml.Serialization;
+using Xunit;
+
+namespace MonoTests.System.Drawing
+{
+
+    public class TestBitmap
+    {
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void TestPixels()
+        {
+            // Tests GetSetPixel/SetPixel			
+            Bitmap bmp = new Bitmap(100, 100, PixelFormat.Format32bppRgb);
+            bmp.SetPixel(0, 0, Color.FromArgb(255, 128, 128, 128));
+            Color color = bmp.GetPixel(0, 0);
+
+            Assert.Equal(Color.FromArgb(255, 128, 128, 128), color);
+
+            bmp.SetPixel(99, 99, Color.FromArgb(255, 255, 0, 155));
+            Color color2 = bmp.GetPixel(99, 99);
+            Assert.Equal(Color.FromArgb(255, 255, 0, 155), color2);
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void LockBits_32_32_NonIndexedWrite()
```

Comment:
LockBits_Invoke_Success has a pretty exhaustive set of test cases, it probably covers most or all of these `LockBits` overloads.